docs/deployment/ci-cd.md

```diff
@@ -4,114 +4,154 @@ SPDX-FileCopyrightText: Copyright (C) 2023 Opal Health Informatics Group at the
+which can be used to build and deploy the app for another environment.
+When building for the prod environment, an option can be selected to also deploy the app to the stores.
+Choosing this option will build a release version of the iOS and Android apps (in addition to the regular build).
+In the case of iOS, this release app is uploaded to TestFlight for release.
+The Android release app can be uploaded manually to Google Play (WIP to use an action to also do this automatically).
 
 To build the app on demand, go to the [GitHub Actions Tab](https://github.com/opalmedapps/opal-app/actions),
-select `Build App` in the left sidebar, and next to `This workflow has a workflow_dispatch event trigger`,
+select `Build and Deploy App` in the left sidebar, and next to `This workflow has a workflow_dispatch event trigger`,
 select `Run workflow`.
-This will open a panel in which to provide inputs, such as which environment to use.
+This will open a panel in which to provide inputs, such as which environment to use and whether to deploy the resulting app.
 
-## Workflows - GitLab
+### Build and Deploy Web
 
-### 1. Commits on feature branches (regular development)
+Similarly to `Build and Deploy App`, the `Build and Deploy Web` workflow runs automatically on all commits to the main branch.
+In addition, it runs on all pull-request and merge-queue commits,
+to ensure that the web build can complete successfully before merging new code.
 
-When working on feature branches, the commits that developers make and push will trigger a basic workflow.
-Specifically, any commits on branches _other_ than the default branch are handled as follows:
+This workflow is constructed similarly to `Build and Deploy App`, except that it builds for the web instead of for a mobile app platform,
+and deploys the site to our web app server.
 
-- Pipeline runs on the pushed commit
-  - Stage 1: `check`
-    - `check dependencies`: The npm dependencies are installed, as a smoke test.
-      This provides early feedback (in the form of a failed pipeline) to alert developers that the build will later fail
-      since the dependencies are unable to install successfully.
-  - Stage 2: `build`
-    - `build android`/`build ios`/`build web` (manual/manual/auto): Mobile app builds are not automatically executed, to save on resources.
-      This makes sense given that developers will rarely need to build the mobile app on every commit they push.
-      However, a manual button is available to build the app on demand (using the Dev environment settings and backend).
-      The resulting `apk`, `ipa` and web files are available as artifacts for download.
-  - Stage 3: `test`
-    - _More details to be added._
+### Increment Version
 
-### 2. Commits on the default branch (after completing merge requests)
+`Increment Version` is a manual-only workflow that runs [semantic-release](https://github.com/semantic-release/semantic-release)
```

Comment:
What would the workflow/process be if someone wanted to release a new version to the app stores (in terms of the order of workflows to execute.

Reply:
I've added a mention of `Increment Version` to the `Quick Guide`, so all the steps should be listed there now.